### PR TITLE
Add template-specific wait timers

### PR DESCRIPTION
**Pull request by slackpad**
Does a rebase on the work from @tgwizard on #589 and adds some tests. While I was in here I also de-nested the wait logic a tiny bit.

/cc @sethvargo

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -642,6 +650,7 @@ type ConfigTemplate struct {
 	Backup         bool          `json:"backup" mapstructure:"backup"`
 	LeftDelim      string        `json:"left_delimiter" mapstructure:"left_delimiter"`
 	RightDelim     string        `json:"right_delimiter" mapstructure:"right_delimiter"`
+	Wait           *watch.Wait   `json:"wait" mapstructure:"wait"`
```
Comment (sean-) [inline config.go]: Is there anything guarding against a negative wait time, or that `max` is less than `min`?  I didn't see anything in this diff that indicated those inputs were tested.

Comment (slackpad) [inline config.go]: Yep - those are covered down in here - https://github.com/hashicorp/consul-template/blob/master/watch/wait_test.go.

File: README.md [hunk 2]
```diff
@@ -291,6 +291,14 @@ template {
...
   left_delimiter  = "{{"
   right_delimiter = "}}"
+
+  // This is the `minimum(:maximum)` to wait before rendering a new template to
+  // disk and triggering a command, separated by a colon (`:`). If the optional
+  // maximum value is omitted, it is assumed to be 4x the required minimum value.
+  // This is a numeric time with a unit suffix ("5s"). There is no default value.
+  // The wait value for a template takes precedence over any globally-configured
+  // wait.
+  wait = "2s:6s"
```
Comment (sethvargo) [inline README.md]: ❤️  Great docs here!

File: watch/wait.go [hunk 3]
```diff
@@ -59,6 +59,11 @@ func ParseWait(s string) (*Wait, error) {
 	return &Wait{min, max}, nil
 }
 
+// IsActive returns true if this wait is active (non-zero).
+func (w Wait) IsActive() bool {
```
Comment (sethvargo) [inline watch/wait.go]: Shouldn't this be `w *Wait`?

**Comment by sethvargo** [comment]
This looks good to me!

**Comment by tgwizard** [comment]
Great, huge thanks @slackpad and @sethvargo!

**Pull request merged**